cmd/deploy.go

```diff
@@ -96,27 +96,29 @@ The command performs the following steps:
 				return nil
 			}
 
-			// Check for the Appsody Operator
-			operatorExists, existingNamespace, operatorExistsErr := operatorExistsWithWatchspace(config.LoggingConfig, namespace, config.Dryrun)
-			if operatorExistsErr != nil {
-				return operatorExistsErr
-			}
-
-			//kargs := []string{"service/appsody-operator"}
-			//_, err := KubeGet(kargs)
-			// Performing the kubectl apply
-			if !operatorExists {
-				config.Debug.logf("Failed to find Appsody operator that watches namespace %s. Attempting to install...", namespace)
-				operatorConfig := &operatorCommandConfig{config.RootCommandConfig, namespace}
-				operatorInstallConfig := &operatorInstallCommandConfig{operatorCommandConfig: operatorConfig}
-				//	operatorInstallConfig.RootCommandConfig = operatorConfig.RootCommandConfig
-				err := operatorInstall(operatorInstallConfig)
-				if err != nil {
-					return errors.Errorf("Failed to install an Appsody operator in namespace %s watching namespace %s. Error was: %v", namespace, namespace, err)
+			if !config.noOperatorInstall {
+				// Check for the Appsody Operator
+				operatorExists, existingNamespace, operatorExistsErr := operatorExistsWithWatchspace(config.LoggingConfig, namespace, config.Dryrun, config.noOperatorCheck)
+				if operatorExistsErr != nil {
+					return operatorExistsErr
 				}
-			} else {
-				config.Debug.logf("Operator exists in %s, watching %s ", existingNamespace, namespace)
 
+				//kargs := []string{"service/appsody-operator"}
+				//_, err := KubeGet(kargs)
+				// Performing the kubectl apply
+				if !operatorExists {
```

Comment:
@Kamran64 - there's this corner scenario that I'd like you to test if possible. Say you are running `appsody deploy` on namespace `xyz`. The logic now checks whether there's an operator in `xyz` that watches `xyz` (in line 101). If we don't find one, we install it. But what about the case in which there's an operator in `xyz` that watches `abc`? The query in line 101 would report no matches, because the operator doesn't watch `xyz`. We'd then try to install the operator in `xyz` again (this time watching `xyz`). Would that work? Fail?

Reply:
@chilanti - I tested this just now:

I installed an operator to namespace `xyz` watching `abc`. I then attempted to deploy my project with a target namespace of `xyz` (and had the `no-operator-check` flag set) I.e. `appsody deploy --namespace xyz --no-operator-check`.

As you said, this attempts to install the operator to `xyz` whilst watching `xyz` but the command fails because an operator already exists in namespace `xyz`. Full error:
```
[Error] Failed to install an Appsody operator in namespace xyz watching namespace xyz. Error was: An operator already exists in namespace xyz and it is watching the abc namespace.
```
I reattempted without the `--no-operator-check` flag and saw the same error.

Reply:
Thanks for testing this! 
I think we should handle the situation. If there's an operator in `xyz`, but it's watching another namespace, we should error out with a message that says "There's an appsody operator in xyz, but it's watching another namespace... talk to your cluster adminstrator".

Reply:
Added in the latest commit - See `operator_install.go` line 86. Thanks @chilanti :)